cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -741,7 +799,7 @@ private StreamSizeScheduleTask(Id.Program programId, SchedulableProgramType prog
      * @param persist {@code true} if this information should be persisted, {@code false} otherwise
      */
     public void startSchedule(long basePollSize, long basePollTs, long lastRunSize, long lastRunTs,
-                              boolean active, boolean persist) {
+                              boolean active, boolean persist) throws SchedulerException {
```

Comment:
similar to my above comment, it is much clearer if you separate the code for restoring an existing schedule task vs. creating a new one